htdocs/core/class/commonobject.class.php

```diff
@@ -3004,57 +3004,56 @@ function fetchObjectLinked($sourceid=null,$sourcetype='',$targetid=null,$targett
-						$classpath = 'supplier_proposal/class';
-					}
-					else if ($objecttype == 'shipping')			{
-						$classpath = 'expedition/class'; $subelement = 'expedition'; $module = 'expedition_bon';
-					}
-					else if ($objecttype == 'delivery')			{
-						$classpath = 'livraison/class'; $subelement = 'livraison'; $module = 'livraison_bon';
-					}
-					else if ($objecttype == 'invoice_supplier' || $objecttype == 'order_supplier')	{
-						$classpath = 'fourn/class'; $module = 'fournisseur';
-					}
-					else if ($objecttype == 'fichinter')			{
-						$classpath = 'fichinter/class'; $subelement = 'fichinter'; $module = 'ficheinter';
-					}
-					else if ($objecttype == 'subscription')			{
-						$classpath = 'adherents/class'; $module = 'adherent';
-					}
-
-					// Set classfile
-					$classfile = strtolower($subelement); $classname = ucfirst($subelement);
-
-					if ($objecttype == 'order') {
-						$classfile = 'commande'; $classname = 'Commande';
-					}
-					else if ($objecttype == 'invoice_supplier') {
-						$classfile = 'fournisseur.facture'; $classname = 'FactureFournisseur';
-					}
-					else if ($objecttype == 'order_supplier')   {
-						$classfile = 'fournisseur.commande'; $classname = 'CommandeFournisseur';
-					}
-					else if ($objecttype == 'supplier_proposal')   {
-						$classfile = 'supplier_proposal'; $classname = 'SupplierProposal';
-					}
-					else if ($objecttype == 'facturerec')   {
-						$classfile = 'facture-rec'; $classname = 'FactureRec';
-					}
-					else if ($objecttype == 'subscription')   {
-						$classfile = 'subscription'; $classname = 'Subscription';
-					}
+					$classfile = strtolower($subelement);
+					$classname = ucfirst($subelement);
+
+					// To work with non standard classpath, classfile or module name
+                    if ($objecttype == 'order') {
+                        $classfile = 'commande';
+                        $classname = 'Commande';
+                    } elseif ($objecttype == 'facture') {
+                        $classpath = 'compta/facture/class';
+                    } elseif ($objecttype == 'facturerec') {
+                        $classpath = 'compta/facture/class';
+                        $module = 'facture';
+                        $classfile = 'facture-rec';
+                        $classname = 'FactureRec';
+                    } elseif ($objecttype == 'propal') {
+                        $classpath = 'comm/propal/class';
+                    } elseif ($objecttype == 'supplier_proposal') {
+                        $classpath = 'supplier_proposal/class';
+                        $classfile = 'supplier_proposal';
+                        $classname = 'SupplierProposal';
+                    } elseif ($objecttype == 'shipping') {
+                        $classpath = 'expedition/class';
+                        $subelement = 'expedition';
+                        $module = 'expedition_bon';
+                    } elseif ($objecttype == 'delivery') {
```

Comment:
Really same result ?
If objecttype = 'delivery', it seems classfile at end of code will differs...

Reply:
'''} else if ($objecttype == 'delivery')	 {
$classpath = 'livraison/class'; $subelement = 'livraison'; $module = 'livraison_bon';
}
'''
'''
} elseif ($objecttype == 'delivery') {
    $classpath = 'livraison/class';
    $subelement = 'livraison';
    $module = 'livraison_bon';
}
'''

Reply:
Yes but few lines after there is also a 

$classfile = strtolower($subelement); $classname = ucfirst($subelement);

so at bottom of code, value differs